src/TranslationUi/Term/Field/MetaDescription.php

```diff
@@ -0,0 +1,78 @@
+<?php # -*- coding: utf-8 -*-
+/*
+ * This file is part of the MultilingualPress Yoast Seo Sync package.
+ *
+ * (c) Inpsyde GmbH
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Inpsyde\MultilingualPress\YoastSeoSync\TranslationUi\Term\Field;
+
+use Inpsyde\MultilingualPress\TranslationUi\MetaboxFieldsHelper;
+use Inpsyde\MultilingualPress\TranslationUi\Term\RelationshipContext;
+use Inpsyde\MultilingualPress\YoastSeoSync\TranslationUi\Term\MetaboxFields;
+
+class MetaDescription
+{
+    /**
+     * @param MetaboxFieldsHelper $helper
+     * @param RelationshipContext $context
+     */
+    public function __invoke(MetaboxFieldsHelper $helper, RelationshipContext $context)
+    {
+        $id = $helper->fieldId(MetaboxFields::FIELD_META_DESCRIPTION);
+        $name = $helper->fieldName(MetaboxFields::FIELD_META_DESCRIPTION);
+        $value = $this->value($context);
+        ?>
+        <tr>
+            <th scope="row">
+                <label>
+                    <?php esc_html_e('Meta description', 'multilingualpress-yoast-seo-sync') ?>
+                </label>
+            </th>
+            <td>
+                <textarea
+                    name="<?= esc_attr($name) ?>"
+                    id="<?= esc_attr($id) ?>"
+                    rows="3"
+                    class="large-text"><?= wp_kses_post($value) ?></textarea>
+            </td>
+        </tr>
+        <?php
+    }
+
+    /**
+     * @param string $value
+     * @return string
+     */
+    public static function sanitize(string $value): string
+    {
+        return sanitize_textarea_field($value);
+    }
+
+    /**
+     * Retrieve the value for the input field.
+     *
+     * @param RelationshipContext $relationshipContext
+     * @return string
+     */
+    private function value(RelationshipContext $relationshipContext): string
+    {
+        $option = get_blog_option($relationshipContext->remoteSiteId(), 'wpseo_taxonomy_meta');
+        if(!$option) {
+            return '';
+        }
+
+        $term = get_term($relationshipContext->remoteTermId());
+        if($term instanceof \WP_Error) {
+            return '';
+        }
+        $taxonomy = $term->taxonomy;
+
+        return $option[$taxonomy][$relationshipContext->remoteTermId()]['wpseo_desc'] ?? '';
+    }
```

Comment:
I see that all of the methods for retrieving the value are practically the same the only difference is the key for which retrieve the value for example in this case 'wpseo_desc'

What about creating one single class `Repository` inside `src/TranslationUi/Term` which has a method that get passed the `RelationshipContext` and the `key` for which option you want to retrieve the value?

```php
class Repository 
{
    const OPTION_KEY = 'wpseo_taxonomy_meta';
  
    public function optionByContext(RelationshipContext $relationshipContext, string $optionKey): string
    {
        if(!$optionKey) {
             throw new \InvalidArgumentException('Option key cannot be empty.');
        }
    
        $option = get_blog_option($relationshipContext->remoteSiteId(), self::OPTION_KEY);
        if(!$option) {
            return '';
        }

        $term = get_term($relationshipContext->remoteTermId());
        if($term instanceof \WP_Error) {
           // Not sure about the name of the method here.
            throw new \DomainException($term->get_error_message());
        }

        return $option[$term->taxonomy][$term->term_id][$optionKey] ?? '';
    }
}
```

Now you have to only inject the class instance to the field classes and you have just to call the method with the right parameters. You'll have one single place for the same logic which will make the solution more maintainable.

Ps. Are you sure the `get_term` is calling from within the target site?